### ISSUE TITLE
Two more classes needed for FHIR DateTime primitive types.

### ISSUE THREAD

**Issue by nazrulworld** [mmabey/fhirdatetime#1]
As we know that three primitive types related to date time are defined in FHIR specification. I think single class `FhirDateTime` is not capable to handle, especially validation issues during construct objects. For instance, the Date type should  only accept datetime.date or string like ´year-month-day´ or `year-month` or `year` on the other hand `Instant` type input must be datetime.datetime with timezone info or ISO formatted string with of course timezone included.

**Comment by mmabey** [comment]
While I appreciate your efforts to find ways to make the library more useful, only the `dateTime` and `date` objects from the FHIR v4 spec make sense for the purposes of this library, which is to allow for the specification of ambiguous dates that may not have the precision required by the native Python types. The reason for this is most evident when you look at the [documentation for the `instant` object](https://hl7.org/fhir/R4/datatypes.html#instant), which says, in part:

> Note: This is intended for when precisely observed times are required (typically system logs etc.), and not human-reported times - for those, use date or dateTime.

In other words, only the `date` and `dateTime` objects are intended for use with human-reported times, meaning they need to support ambiguity and lack of precision. That is the purpose of this library. Any other data structures relating to date or time should be mapped to their Python-native types.
